pkg/packing/packing.go

```diff
@@ -70,77 +62,54 @@ func (p *packer) Pack(ctx context.Context, pods []*v1.Pod, instanceTypes []*Inst
 	var packings []*Packing
 	var packing *Packing
 	remainingPods := pods
-	nodeCapacities := p.getNodeCapacities(instanceTypes, constraints)
 	for len(remainingPods) > 0 {
-		packing, remainingPods = p.packWithLargestPod(remainingPods, nodeCapacities)
-		// checked all instance type and found no packing option
+		packing, remainingPods = p.packWithLargestPod(remainingPods, instanceTypes, constraints)
+		// checked all instance types and found no packing option
 		if len(packing.Pods) == 0 {
-			zap.S().Warnf("Failed to find viable instance type for pod %s/%s ", remainingPods[0].Namespace, remainingPods[0].Name)
+			zap.S().Warnf("Failed to compute packing for pod(s) %v with instance type option(s) %v", podNames(remainingPods), instanceTypeNames(instanceTypes))
 			remainingPods = remainingPods[1:]
 			continue
 		}
 		packings = append(packings, packing)
-		sortByResources(packing.InstanceTypes)
-		instanceTypeNames := []string{}
-		for _, it := range packing.InstanceTypes {
-			instanceTypeNames = append(instanceTypeNames, *it.InstanceType)
-		}
-		zap.S().Debugf("Selected %d instance type options for %d pod(s) %v", len(packing.InstanceTypes), len(packing.Pods), instanceTypeNames)
+		zap.S().Debugf("Successfully computed packing for pod(s) %v with instance type option(s) %s", podNames(packing.Pods), instanceTypeNames(packing.InstanceTypes))
 	}
 	return packings
 }
 
-func (*packer) getNodeCapacities(instanceTypes []*Instance, constraints *cloudprovider.Constraints) []*nodeCapacity {
-	nodeCapacities := []*nodeCapacity{}
-	for _, instanceType := range instanceTypes {
-		nc := nodeCapacityFrom(instanceType)
-		kubeletOverhead := binpacking.CalculateKubeletOverhead(nc.total)
-		if ok := nc.reserve(resources.Merge(constraints.Overhead, kubeletOverhead)); !ok {
-			zap.S().Infof("Excluding instance type %s because there are not enough resources for the kubelet overhead", nc.instanceType)
-			continue
-		}
-		nodeCapacities = append(nodeCapacities, nc)
-	}
-	return nodeCapacities
-}
-
 // packWithLargestPod will try to pack max number of pods with largest pod in
 // pods across all available node capacities. It returns Packing: max pod count
```

Comment:
nit: "packWithLargestPod will try to pack the max number of pods starting with the largest pod across all available node capacities"